Makefile

```diff
@@ -80,7 +80,7 @@ style-fix:
 		echo "Refusing to run on dirty git state. Commit your changes first."; \
 		exit 1; \
 	fi; \
-	git ls-files | grep -e '\.\(c\|h\|cpp\)$$' | xargs clang-format -i -style=file;
+	git ls-files | grep -e '\(\.*\.c$$\)\|\(\.*\.h$$\)\|\(\.*\.cpp$$\)\&\(\!\(\^tests\/\)\(\.*\)\)' | xargs clang-format -i -style=file;
```

Comment:
I'd split this now rather big grep call into two separate steps. Something like: `git ls-files | grep -e '\.\(c\|h\|cpp\)$'|grep -v '^tests/'` (May need some escaping for it to work in the Makefile.